packages/block-editor/src/components/block-list/use-in-between-inserter.js

```diff
@@ -107,6 +107,15 @@ export function useInBetweenInserter() {
 					}
 				}
 
+				// Don't show the insertion point if a parent block has an "overlay"
+				// See https://github.com/WordPress/gutenberg/pull/34012#pullrequestreview-727762337
+				const parentOverlay = element.closest(
+					'.block-editor-block-content-overlay.overlay-active'
+				);
```

Comment:
A bit hacky but I'm not sure there's a better solution.

Reply:
Yeah!  I originally tried something similar to this before moving to using the extra element, and figured the extra element was preferable to the ad-hoc hackyness of conditionally disabling the inserter.  If you think this is more preferable than the added element, then it seems like a fair approach to me. 

However, I am still finding some inserters on edge cases.  Testing the nested template-part case, oddly an inserter appears just above the nested template part:

![nested-TP-inserter-goo](https://user-images.githubusercontent.com/28742426/129232905-abfb743a-715d-4923-8736-73f42e546d6b.gif)